### PR TITLE
Added idletime settings for homunculus idle no share

### PR DESCRIPTION
**Pull request by Daegaladh**
* Follow up to 921b3e2c3e9eae36cbf2ca35bcede6e0d73ba074

* **Addressed Issue(s)**: N/A

* **Server Mode**: Both

* **Description of Pull Request**: Adds idletime settings for homunculus idle no share separated from the party share idletime settings.

### CODE REVIEW

**Comment by Daegaladh** [comment]
Probably fixes #4756

File: src/map/script.cpp [hunk 1]
```diff
@@ -17385,6 +17385,17 @@ BUILDIN_FUNC(checkidle)
 	return SCRIPT_CMD_SUCCESS;
 }
 
+BUILDIN_FUNC(checkidlehom)
+{
+	TBL_PC *sd = NULL;
+
+	if( script_nick2sd(2,sd) )
+		script_pushint(st, DIFF_TICK(last_tick, sd->idletime_hom));
```
Comment (Lemongrass3110) [inline src/map/script.cpp:17393]: Shouldnt this simply return pc_isidle_hom?

File: src/map/battle.cpp [hunk 2]
```diff
@@ -8875,6 +8875,7 @@ static const struct _battle_data {
 	{ "boss_nopc_idleskill_rate",           &battle_config.boss_nopc_idleskill_rate,        100,    0,    100,              },
 	{ "boss_nopc_move_rate",                &battle_config.boss_nopc_move_rate,             100,    0,    100,              },
 	{ "hom_idle_no_share",                  &battle_config.hom_idle_no_share,               0,      0,      INT_MAX,        },
+	{ "idletime_hom_option",                &battle_config.idletime_hom_option,             0x25,   1,      INT_MAX,        },
```
[the remark this answers is not in the document]
Comment (Daegaladh) [reply src/map/battle.cpp:8878]: I just copypasted from the normal idletime_option and didn't think much of it, I'll fix both

File: src/map/script.cpp (hunk 1 above)
Comment (Daegaladh) [reply src/map/script.cpp:17393]: No, that would only return if the player is idle or not, not the idle time

**Pull request merged**